### PR TITLE
configobserver: add helper function for observing the Proxy object

### PR DESCRIPTION
**Pull request by stlaz**
OpenShift infrastructure operators should be observing the Proxy object, add a helper so that we can reuse the code.

Useful at least for the kube and openshift apiserver operators.

### CODE REVIEW

**Comment by stlaz** [comment]
Added a unit test, removed logging on Proxy object not-foud to match the featuregates observer behavior.

File: pkg/operator/configobserver/proxy/observe_proxy.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		configPath: configPath,
+	}).ObserveProxyConfig
+}
+
+type observeProxyFlags struct {
+	configPath []string
+}
+
+// ObserveProxyConfig observes the proxy.config.openshift.io/cluster object and writes
+// its content an unstructured object in a string map at the proxyObserver.proxy path
```
Comment (sttts) [inline pkg/operator/configobserver/proxy/observe_proxy.go]: nit: as an unstructured object

Comment (sttts) [inline pkg/operator/configobserver/proxy/observe_proxy.go]: s/proxyObserver.proxy path/at the path given to the constructor/

File: pkg/operator/configobserver/proxy/observe_proxy.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+func (f *observeProxyFlags) ObserveProxyConfig(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	proxyLister := genericListers.(ProxyLister)
+
+	errs := []error{}
```
Comment (sttts) [inline pkg/operator/configobserver/proxy/observe_proxy.go]: the use of errs is not helpful in the flow here, even wrong (see after the `Get("cluster")`. Would switch to a simple err.

Comment (stlaz) [reply pkg/operator/configobserver/proxy/observe_proxy.go]: This is how the errors are being handled throughout the other observers as well (see `observer_cloudprovider.go` or `observe_featuregates.go`) - they only return the errors that occured during handling of the unstructured object.

I agree that swallowing the error at the `Get("cluster")` part is not optimal, I'll use the even recorder similarly to what's done in `ObserveCloudProviderNames()`

Comment (stlaz) [reply pkg/operator/configobserver/proxy/observe_proxy.go]: Hm, actually, that will only suffice for the `NotFound()` way, but the one with actual error should be added, that could be a problem even for the `clouprovider` observing, I'll fix it there as well.

File: pkg/operator/configobserver/proxy/observe_proxy.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+	}
+
+	observedConfig := map[string]interface{}{}
+	proxyConfig, err := proxyLister.ProxyLister().Get("cluster")
+	if errors.IsNotFound(err) {
+		recorder.Warningf("ObserveProxyConfig", "proxy.%s/cluster not found", configv1.GroupName)
+		return observedConfig, errs
+	}
+	if err != nil {
+		errs = append(errs, err)
```
Comment (sttts) [inline pkg/operator/configobserver/proxy/observe_proxy.go]: :+1:

**Pull request merged**